### PR TITLE
Do not fail when bcolz import fails

### PR DESCRIPTION
**Pull request by benjello**

### CODE REVIEW

**Comment by gdementen** [comment]
If you want to make bcolz optional (which is fine with me given its very limited usage), please include an explicit exception if its functionality (interpolate in this case) is used anyway.

2 other places need to be adapted too:
* PrintVersionsAction in main.py 
https://github.com/benjello/liam2/blob/master/src/main.py#L171
* https://github.com/liam2/liam2/blob/master/INSTALL

**Comment by benjello** [comment]
Just starting the conversation to test your mood ;-)
More commits to come. Please do not hesitate to be straightforward in your comments, it actually helps me understand your priorities.

**Comment by benjello** [comment]
Could you write the sentence you would like to appear in https://github.com/liam2/liam2/blob/master/INSTALL ?

**Comment by gdementen** [comment]
just move it to the optional dependencies paragraph with something like:

`to import data with interpolation/data with missing data points (eg several time series with different dates for the same individual):`

File: src/main.py [hunk 1]
```diff
@@ -172,7 +172,10 @@ class PrintVersionsAction(argparse.Action):
     def __call__(self, parser, namespace, values, option_string=None):
         import numpy
         import numexpr
-        import bcolz
+        try:
+            import bcolz
+        except ImportError:
+            bcolz = None
```
Comment (gdementen) [inline src/main.py]: plz move this below the "optional deps" comment

File: src/main.py [hunk 2]
```diff
@@ -210,7 +213,7 @@ def __call__(self, parser, namespace, values, option_string=None):
 matplotlib {mpl}
 """.format(py=py_version, np=numpy.__version__, ne=numexpr.__version__,
            pt=tables.__version__, vt=vt_version, mpl=mpl_version,
-           bc=bcolz.__version__, yml=yaml.__version__))
+           bc=bcolz.__version__ if bcolz is not None else None, yml=yaml.__version__))
```
Comment (gdementen) [inline src/main.py]: plz use N/A when missing, and use the same template (bc_version = ...) than for other optional deps (it is not especially better, but lets be consistent)

File: src/setup.py [hunk 3]
```diff
@@ -143,5 +143,12 @@ def vitables_data_files():
       ext_modules=ext_modules,
       options={"build_ext": build_ext_options, "build_exe": build_exe_options},
       executables=[Executable("main.py")],
-      requires=['numpy', 'numexpr', 'tables', 'bcolz'])
+      install_requires=['numpy', 'numexpr', 'tables'],
+      extra_requires=dict(
```
Comment (gdementen) [inline src/setup.py]: Thanks for this! I never bothered to look up the exact syntax :)

Comment (benjello) [inline src/setup.py]: Not tested though !

Comment (gdementen) [inline src/setup.py]: Please test then ;-)

Comment (benjello) [inline src/setup.py]: I actually use another setup.py hacked to work on a linux machine so won't be able to test it for windows users (@AlexisEidelman ?). I do not plan to submit a complete overhaul of setup.py (with a travis CI config file etc) this early. My pint with this commit was to propagate upstream as soon as possible what could be propagated without to much hassle.
But i did checked the syntax though ;-)

Comment (gdementen) [inline src/setup.py]: hmmm a quick google seem to suggest "extras_require" instead of "extra_requires"...

Comment (benjello) [inline src/setup.py]: Sorry I will fix that

**Comment by gdementen** [comment]
Looks good to me, thanks!

**Pull request merged**